File: src/cod/advantages.rs
#![allow(non_snake_case, non_upper_case_globals)]

use serde::{
	Serialize,
	Deserialize,
};
use std::iter::Iterator;
use strum_macros::{
	AsRefStr,
	EnumCount,
	EnumIter
};
use crate::cod::tracks::Tracker;

pub const BaseSpeed: usize = 5;

/// The Advantages of a Chronicles of Darkness character.
#[derive(AsRefStr, Clone, Copy, Debug, Deserialize, EnumCount, EnumIter, Eq, PartialEq, Serialize, PartialOrd, Ord)]
pub enum CoreAdvantageType
{
	Defense,
	Health,
	Initiative,
	//Integrity(String),
	Size,
	Speed,
	//Vice(String),
	//Virtue(String),
	Willpower,
}

/// Data structure defining the Advantages of a Chronicles of Darkness character.
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct CoreAdvantages
{
	#[serde(default)]
	pub defense: usize,
	#[serde(default)]
	pub health: Tracker,
	#[serde(default)]
	pub initiative: usize,
	#[serde(default)]
	pub size: usize,
	#[serde(default)]
	pub speed: usize,
	#[serde(default)]
	pub willpower: Tracker,
}

impl Default for CoreAdvantages
{
	fn default() -> Self
	{
		Self
		{
			defense: 1,
			health: Tracker::new(6),
			initiative: 2,
			size: 5,
			speed: 7,
			willpower: Tracker::new(2),
		}
	}
}

File: OCSM/cofd/Advantages.cs

namespace Ocsm.Cofd;

public class Advantages()
{
	public int Armor { get; set; }
	public int Defense { get; set; }
	public Health Health { get; set; } = new Health();
	public int Initiative { get; set; }
	public int Integrity { get; set; } = 7;
	public int Power { get; set; } = 1;
	public int ResourceSpent { get; set; }
	public int ResourceMax { get; set; } = 10;
	public int Size { get; set; } = 5;
	public int Speed { get; set; }
	public int WillpowerSpent { get; set; }
	public int WillpowerMax { get; set; } = 2;
}
